File: app/src/main/java/ms/imagine/foodiemate/presenters/FbAuthStatePresenter.java
package ms.imagine.foodiemate.presenters;

import android.util.Log;
import com.facebook.login.LoginManager;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;


public class FbAuthStatePresenter {
    FirebaseAuth mAuth;  //usernode here


    public FbAuthStatePresenter(){
        mAuth = FirebaseAuth.getInstance();
    }

    public FirebaseUser userState() {
        return mAuth.getCurrentUser();
    }

    public void signOut(){
        mAuth.signOut();
        FirebaseAuth.getInstance().signOut();
        LoginManager.getInstance().logOut();
        Log.w("SIGNs", "not null MAuth");
    }
}

File: app/src/main/java/ms/imagine/foodiemate/Presenter/FbAuthStatePresenter.kt
package ms.imagine.foodiemate.Presenter

import android.util.Log
import com.facebook.login.LoginManager
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.auth.FirebaseUser


class FbAuthStatePresenter {
    var mAuth: FirebaseAuth;  //usernode here


    constructor(){
        mAuth = FirebaseAuth.getInstance()
    }

    fun userState(): FirebaseUser? {
        return mAuth.currentUser
    }

    fun signOut(){
        mAuth.signOut()
        FirebaseAuth.getInstance().signOut()
        LoginManager.getInstance().logOut()
        Log.w("SIGNs", "not null MAuth")
    }
}